src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -102,20 +106,42 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
         IndexConstants.INDEX_NUM_BUCKETS,
         IndexConstants.INDEX_NUM_BUCKETS_DEFAULT.toString)
       .toInt
-    val selectedColumns = indexConfig.indexedColumns ++ indexConfig.includedColumns
+
+    val (resolvedIndexedColumns, resolvedIncludedColumns) = resolveConfig(df, indexConfig)
+    val selectedColumns = resolvedIndexedColumns ++ resolvedIncludedColumns
     val indexDataFrame = df.select(selectedColumns.head, selectedColumns.tail: _*)
-    val indexColNames = indexConfig.indexedColumns
 
     // run job
     val repartitionedIndexDataFrame =
-      indexDataFrame.repartition(numBuckets, indexColNames.map(df(_)): _*)
+      indexDataFrame.repartition(numBuckets, resolvedIndexedColumns.map(df(_)): _*)
 
     // Save the index with the number of buckets specified.
     repartitionedIndexDataFrame.write
       .saveWithBuckets(
         repartitionedIndexDataFrame,
         indexDataPath.toString,
         numBuckets,
-        indexColNames)
+        resolvedIndexedColumns)
+  }
+
+  private def resolveConfig(
+      df: DataFrame,
+      indexConfig: IndexConfig): (Seq[String], Seq[String]) = {
+    val spark = df.sparkSession
+    val dfColumnNames = df.schema.fieldNames
+    val resolvedIndexedColumns =
+      ResolverUtils.resolve(spark, indexConfig.indexedColumns, dfColumnNames)
+    val resolvedIncludedColumns =
+      ResolverUtils.resolve(spark, indexConfig.includedColumns, dfColumnNames)
+
+    val allColumns = resolvedIndexedColumns ++ resolvedIncludedColumns
+
+    (resolvedIndexedColumns, resolvedIncludedColumns) match {
+      case (Some(indexed), Some(included)) => (indexed, included)
+      case _ =>
+        throw HyperspaceException(
+          s"Unexpected Exception: Some of the columns from $allColumns could not be resolved " +
```

Comment:
Can we pinpoint the exact column that failed? Since this is during create, it will be helpful for the user.

Reply:
thanks @imback82, added